packages/typespec-ts/src/transform/transformResponses.ts

```diff
@@ -141,6 +142,9 @@ function transformHeaders(
         SchemaContext.Output
       ]) as Schema;
       const type = getTypeName(typeSchema);
+      if (typeSchema.type === "object" && typeSchema.enum) {
```

Comment:
I wonder it it would be helpful to create a helper util to check if the schema has an enum